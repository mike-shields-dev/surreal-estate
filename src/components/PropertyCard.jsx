import React from "react";
import PropTypes from "prop-types";
import { HiOutlineMail } from "react-icons/hi";
import { FaFortAwesome, FaBath, FaBed } from "react-icons/fa";
import styles from "../styles/PropertyCard.module.css";

const PropertyCard = ({
  bathrooms,
  bedrooms,
  city,
  email,
  price,
  title,
  type,
}) => {
  return (
    <div className={styles["property-card"]}>
      <header className={styles["property-card__header"]}>
        <FaFortAwesome title="surreal estate logo" />
      </header>
      <main className={styles["property-card__main"]}>
        <h2 className={styles["property-card__title"]}>{title}</h2>
        <p>
          <span className={styles["property-card__type-location"]}>{type}</span>{" "}
          - <span>{city}</span>
        </p>
        <p className={styles["property-card__bathrooms"]}>
          <FaBath
            className={styles["property-card__bath-icon"]}
            title="bath icon"
          />{" "}
          <span>{bathrooms}</span>
        </p>
        <p className={styles["property-card__bedrooms"]}>
          <FaBed
            className={styles["property-card__bed-icon"]}
            title="bed icon"
          />{" "}
          <span>{bedrooms}</span>
        </p>
        <p>
          <span className={styles["property-card__currency"]}>Â£</span> {price}
        </p>
      </main>
      <a className={styles["property-card__mailto"]} href={`mailto:${email}`}>
        <HiOutlineMail className={styles["property-card__mail-icon"]} />
        <span>email</span>
      </a>
    </div>
  );
};

PropertyCard.propTypes = {
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default PropertyCard;
