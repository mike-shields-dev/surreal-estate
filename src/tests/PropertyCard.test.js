import React from "react";
import { render, screen } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

const props = {
  title: "2 bedroom period property",
  type: "Flat",
  bedrooms: 2,
  bathrooms: 3,
  price: 1000,
  city: "Liverpool",
  email: "example@example.com",
};

const renderPropertyCard = () => render(<PropertyCard {...props} />);

describe("PropertyCard", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderPropertyCard();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the surreal estate logo", () => {
    renderPropertyCard();

    expect(screen.getByTitle("surreal estate logo")).toBeInTheDocument();
  });

  it("renders the title prop", () => {
    renderPropertyCard();

    expect(screen.getByText([props.title])).toBeInTheDocument();
  });

  it("renders the type prop", () => {
    renderPropertyCard();

    expect(screen.getByText(props.type)).toBeInTheDocument();
  });

  it("renders the city prop", () => {
    renderPropertyCard();

    expect(screen.getByText(props.city)).toBeInTheDocument();
  });

  it("renders the bathrooms prop", () => {
    renderPropertyCard();

    expect(screen.getByText(props.bathrooms)).toBeInTheDocument();
  });

  it("renders a bath icon", () => {
    renderPropertyCard();

    expect(screen.getByTitle("bath icon")).toBeInTheDocument();
  });

  it("renders the bedrooms prop", () => {
    renderPropertyCard();

    expect(screen.getByText(props.bedrooms)).toBeInTheDocument();
  });

  it("renders a bed icon", () => {
    renderPropertyCard();

    expect(screen.getByTitle("bed icon")).toBeInTheDocument();
  });

  it("renders price prop", () => {
    renderPropertyCard();

    expect(screen.getByText(props.bedrooms)).toBeInTheDocument();
  });

  it("renders a 'mailto' link with the give email prop", () => {
    renderPropertyCard();
    const mailtoLink = screen.getByText(/email/i);

    expect(mailtoLink.closest("a")).toHaveAttribute(
      "href",
      `mailto:${props.email}`
    );
  });
});
