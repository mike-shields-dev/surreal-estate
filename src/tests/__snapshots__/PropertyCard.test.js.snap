// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <header
      class="property-card__header"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          surreal estate logo
        </title>
        <path
          d="M489.2 287.9h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6V146.2c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-6-8-4.6-11.7-4.6v-38c8.3-2 17.1-3.4 25.7-3.4 10.9 0 20.9 4.3 31.4 4.3 4.6 0 27.7-1.1 27.7-8v-60c0-2.6-2-4.6-4.6-4.6-5.1 0-15.1 4.3-24 4.3-9.7 0-20.9-4.3-32.6-4.3-8 0-16 1.1-23.7 2.9v-4.9c5.4-2.6 9.1-8.3 9.1-14.3 0-20.7-31.4-20.8-31.4 0 0 6 3.7 11.7 9.1 14.3v111.7c-3.7 0-11.7-1.4-11.7 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32H128v-32c0-2.6-2-4.6-4.6-4.6H96c-2.6 0-4.6 2-4.6 4.6v178.3H54.8v-32c0-2.6-2-4.6-4.6-4.6H22.8c-2.6 0-4.6 2-4.6 4.6V512h182.9v-96c0-72.6 109.7-72.6 109.7 0v96h182.9V292.5c.1-2.6-1.9-4.6-4.5-4.6zm-288.1-4.5c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64zm146.4 0c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64z"
        />
      </svg>
    </header>
    <main
      class="property-card__main"
    >
      <h2
        class="property-card__title"
      >
        2 bedroom period property
      </h2>
      <p>
        <span
          class="property-card__type-location"
        >
          Flat
        </span>
         - 
        <span>
          Liverpool
        </span>
      </p>
      <p
        class="property-card__bathrooms"
      >
        <svg
          class="property-card__bath-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            bath icon
          </title>
          <path
            d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
          />
        </svg>
         
        <span>
          3
        </span>
      </p>
      <p
        class="property-card__bedrooms"
      >
        <svg
          class="property-card__bed-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 640 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            bed icon
          </title>
          <path
            d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          />
        </svg>
         
        <span>
          2
        </span>
      </p>
      <p>
        <span
          class="property-card__currency"
        >
          Â£
        </span>
         1000
      </p>
    </main>
    <a
      class="property-card__mailto"
      href="mailto:example@example.com"
    >
      <svg
        class="property-card__mail-icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <span>
        email
      </span>
    </a>
  </div>
</DocumentFragment>
`;
